README.md

```diff
@@ -1,149 +1,194 @@
-
-### Docker
-
-`sgrep` is packaged within a [docker container](https://hub.docker.com/r/returntocorp/sgrep), making installation as easy as [installing docker](https://docs.docker.com/install/).
+```
+$ docker pull returntocorp/sgrep
+```
 
-### Mac \(alpha\)
+`sgrep` binaries can also be installed on OSX and Ubuntu.
 
-The brew install is WIP and will change significantly.
+### OSX
 
-```bash
-brew tap returntocorp/sgrep https://github.com/returntocorp/sgrep.git
-brew install sgrep-r2c
-# Due to a naming conflict, when installed via brew, it is named `sgrok`
-sgrok --help
+```
+$ brew tap returntocorp/sgrep https://github.com/returntocorp/sgrep.git
+$ brew install sgrep-r2c
+$ sgrep-lint --help
 ```
 
-## Quickstart
-
-```bash
-docker pull returntocorp/sgrep
-
-cd /path/to/repo
-# generate a template config file
-docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep --generate-config
+### Ubuntu
 
-# look for findings
-docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep
+```
+$ wget https://github.com/returntocorp/sgrep/releases/download/v0.4.9/sgrep-0.4.9-ubuntu-16.04.tgz
+$ wget https://github.com/returntocorp/sgrep/releases/download/v0.4.9/sgrep-0.4.9-ubuntu-16.04.tgz.sha256
+$ sha256sum --check sgrep-0.4.9-ubuntu-16.04.tgz.sha256
+$ tar xf sgrep-0.4.9-ubuntu-16.04.tgz
+$ ./sgrep-lint-files/sgrep-lint --help
 ```
 
-## Usage
-
-### Rule Development
+## Using
 
-To rapidly iterate on a single pattern, you can test on a single file or folder. For example,
+Let's start with a simple example:
 
-```bash
-docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep -l python -e '$X == $X' path/to/file.py
+```
+$ cat << EOF > test.py
+a = 1
+b = 2
+if a == a:  # oops, supposed to be a == b
+    print('sgrep test')
+EOF
 ```
 
-Here, `sgrep` will search the target with the pattern `$X == $X` \(which is a stupid equals check\) and print the results to `stdout`. This also works for directories and will skip the file if parsing fails. You can specifiy the language of the pattern with `--lang javascript` for example.
+```
+$ docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep --lang python --pattern '$X == $X' test.py
+test.py
+3:if a == a:  # oops, supposed to be a == b
+```
 
-To see more options
+From here you can use our rules to search for issues in your codebase:
 
-```bash
-docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep --help
+```
+$ cd /path/to/code
+$ docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep --config https://r2c.dev/default-r2c-checks
 ```
 
-### Config Files
-
-#### Format
+You can also [create your own rules](docs/config/advanced.md):
 
-See [config/advanced.md](docs/config/advanced.md) for example configuration files and details on the syntax.
+```
+$ cd /path/to/code
+$ docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep --generate-config
+$ docker run --rm -v "${PWD}:/home/repo" returntocorp/sgrep
+```
 
-#### sgrep Registry
+## Configuring
```

Comment:
The configuration docs seemed to be jumping around different concepts a lot. Between the registry, rule writing, design philosophy, tool defaults, etc. I think we can consolidate a lot of the information and move the finer points like design philosophy to the advanced documentation.